src/irmin-pack/irmin_pack.ml

```diff
@@ -1026,19 +720,19 @@ module Pack (K : Irmin.Hash.S) = struct
         | exception Not_found -> (
           match Index.find t.pack.index k with
           | None -> None
-          | Some e ->
-              let buf, pos = Pool.read t.pages ~off:e.offset ~len:e.len in
+          | Some (off, len) ->
+              let buf, pos = Pool.read t.pages ~off ~len in
               let hash off =
-                match Hashtbl.find t.pack.index.offsets off with
+                (* match Hashtbl.find t.pack.index.offsets off with
```

Comment:
did we loose the offset cache?

Reply:
In the current version, yes. I can't see any performance difference on my (handmade) tests for now.

Reply:
maybe as a first step before splitting everything I'll vendor index in the repo so that we can try it on tezos dataset more easily.